crates/ty_python_semantic/src/types.rs

```diff
@@ -7650,6 +7651,19 @@ impl<'db> UnionType<'db> {
         Self::from_elements(db, self.elements(db).iter().map(transform_fn))
     }
 
+    /// A fallible version of [`UnionType::map`].
+    pub(crate) fn try_map(
```

Comment:
Maybe: `map_options` and `map_ok` similar to the Itertools naming?

and `from_option_elements` and `from_ok_elements`?

I also think it's important to document what "fallible" means because it wasn't clear to me if it bails if there's any `None` value or if it skip sover `None` values.